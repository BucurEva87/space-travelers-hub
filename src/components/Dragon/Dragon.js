import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import style from './Dragon.module.scss';
import { bookDragon, unbookDragon } from '../../redux/dragons/dragons';

const Dragon = (props) => {
  const {
    id, name, type, img, reserved,
  } = props;
  const dispatch = useDispatch();
  return (
    <div key={id} className={style['dragon-container']}>
      <img className={style['dragon-image']} src={img} alt="Dragon" />
      <div className={style['name-type-container']}>
        { reserved ? (<button type="button" className={style['reserved-tag']}>Reserved</button>) : ''}
        <p className={style.name}>{name}</p>
        <p className={style.type}>{type}</p>
        {
          reserved
            ? (<button type="button" onClick={() => dispatch(unbookDragon(id))} className={style['btn-unreserve']}>Cancel Reservation</button>)
            : (<button type="button" onClick={() => dispatch(bookDragon(id))} className={style['btn-reserve']}>Reserve</button>)

        }

      </div>
    </div>
  );
};

Dragon.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Dragon;
