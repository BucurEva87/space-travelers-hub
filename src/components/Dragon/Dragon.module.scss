@import '../../styles/variables';
.dragon-container {
	display: flex;
	gap: 1.6rem;
	padding-bottom: 2rem;
}

.dragon-container img {
	height: 20vh;
	width: 20%;
	border-radius: 0.4rem;	
}

.name-type-container {
	display: flex;
	flex-direction: column;
	font-family: $montserrat;
	font-size: 2rem;
}

.type {
	font-size: 1.5rem;
}

.btn-reserve {
	background-color: $primaryBtnBlue;
	padding: 0.6rem 1rem;
	border: none;
	color: $white;
	margin-top: 1.5rem;
}

.btn-unreserve {
	background-color: $white;
	padding: 0.4rem 1rem;
	border: 0.1rem solid $gray;
	color: $gray;
	margin-top: 1rem;
}

.reserved-tag {
	padding: 0.1rem;
	background-color: $reservedTag;
	color: $white;
	font-size: 0.8rem;
	font-weight: 600;
	width: 4rem;
	border-radius: 0.3rem;
	border: none;
}

