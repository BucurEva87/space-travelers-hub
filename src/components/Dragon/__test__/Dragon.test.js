import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import Dragon from '../Dragon';

let getAllByRole;

const dragons = [
  {
    id: 'dragon1',
    name: 'Dragon 1',
    type: 'capsule',
    img: 'https://farm8.staticflickr.com/7647/16581815487_6d56cb32e1_b.jpg',
    reserved: false,
  },
];

beforeEach(() => {
  const component = render(
    <Provider store={store}>
      <Dragon
        key={dragons[0].id}
        id={dragons[0].id}
        name={dragons[0].name}
        type={dragons[0].type}
        img={dragons[0].img}
        reserved={dragons[0].reserved}
      />
    </Provider>,
  );

  getAllByRole = component.getAllByRole;
});

describe('Dragon component', () => {
  test('is rendered one time on the screen', () => {
    expect(getAllByRole('article')).toHaveLength(1);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Dragon
          key={dragons[0].id}
          id={dragons[0].id}
          name={dragons[0].name}
          type={dragons[0].type}
          img={dragons[0].img}
          reserved={dragons[0].reserved}
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
