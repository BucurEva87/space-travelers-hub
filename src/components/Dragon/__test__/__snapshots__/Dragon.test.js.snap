// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon component matches snapshot 1`] = `
<article
  className="dragon-container"
>
  <img
    alt="Dragon"
    className="dragon-image"
    src="https://farm8.staticflickr.com/7647/16581815487_6d56cb32e1_b.jpg"
  />
  <div
    className="name-type-container"
  >
    <p
      className="name"
    >
      Dragon 1
    </p>
    <p
      className="type"
    >
      capsule
    </p>
    <button
      className="btn-reserve"
      onClick={[Function]}
      type="button"
    >
      Reserve
    </button>
  </div>
</article>
`;
